### PR TITLE
Rc 3150

### PR DESCRIPTION
**Pull request by prckent**
## Proposed changes

Proposed 3.15.0 release. Branched after the weekend tests.

Note that the changelog was written by hand and should be carefully checked.

## What type(s) of changes does this code introduce?

- Bugfix
- New feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no api changes)
- Build related changes
- Testing changes (e.g. new unit/integration/performance tests)
- Documentation changes
- Other (please describe):

### Does this introduce a breaking change?

- Yes

## What systems has this change been tested on?

Weeklies/Nightlies

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by prckent** [comment]
@jtkrogel Please verify the NEXUS section. Thanks.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,87 @@
...
+* The Quantum ESPRESSO converter, pw2qmcpack is now supported via a plugin
+  activated via -DQE_ENABLE_PLUGINS=pw2qmcpack on the QE CMake configure line,
+  see
+  https://qmcpack.readthedocs.io/en/develop/installation.html#quantum-espresso-7-0
+  . The latest QE 7.1 and earlier 7.0 are supported, and new versions should be
+  automatically compatible.
+* Substantial improvements to the performance portable / batched implementation.
+  Using LLVM 15.0, high performance production calculations can be performed on
+  NVIDIA GPUs for several wavefunction types, in addition to running on all CPU
+  systems. As introduced in v3.14.0, the optional project data input parameter
```
Comment (markdewing) [inline CHANGELOG.md:29]: The "project data input parameter" should be a separate bullet point?

Comment (markdewing) [reply CHANGELOG.md:29]: Or nested bullet points for llvm 15 and driver input parameter?

Comment (prckent) [reply CHANGELOG.md]: Good suggestion. I have now split this part into two bullet points. More readable

**Review by jtkrogel — APPROVED** [review]
Nexus log looks good

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,88 @@
...
+* Pseudopotential non-local channel can be specified in input [\#4032](https://github.com/QMCPACK/qmcpack/pull/4032)
+* Use of deprecated CUDA texture API removed for greater compatibility [\#4022](https://github.com/QMCPACK/qmcpack/pull/4022)
+* Optimization has been removed from the legacy CUDA code. New calculations
+  needing GPU support should use the batched drivers and their GPU capabilities
+  for optimization. [\#4138](https://github.com/QMCPACK/qmcpack/pull/4138) 
+* Initial version of determinant update in SYCL for Intel architectures (e.g.
+  [\#4118](https://github.com/QMCPACK/qmcpack/pull/4118) )
+* Updated walker counts in several of the performance tests. Due to the changed
+  but more representative workloads, new performance timings should not be
+  compared with older runs (e.g. [\4112](https://github.com/QMCPACK/qmcpack/pull/4112) )
```
Comment (markdewing) [inline CHANGELOG.md:54]: Missing the # before 4112

Comment (markdewing) [reply CHANGELOG.md:54]: It still links okay, it just looks different

Comment (ye-luo) [reply CHANGELOG.md]: Fixed it via web.

**Comment by ye-luo** [comment]
Test this please

**Review by markdewing — APPROVED** [review]

**Comment by prckent** [comment]
I ran a fairly complete set of tests with gcc12.1+openblas+openMPI, as used in the nightlies. This included QE, PySCF, and RMG runs.
```
98% tests passed, 52 tests failed out of 2212

Label Time Summary:
QMCPACK                     = 61205.57 sec*proc (332 tests)
QMCPACK-checking-results    = 105.38 sec*proc (1502 tests)
abort                       = 1249.10 sec*proc (24 tests)
bug                         = 5062.80 sec*proc (58 tests)
cause_unknown               = 9619.82 sec*proc (131 tests)
check_fail                  = 1751.21 sec*proc (19 tests)
converter                   = 14659.01 sec*proc (117 tests)
coverage                    = 142.06 sec*proc (7 tests)
definite_stat_fail          = 1445.96 sec*proc (10 tests)
deterministic               = 1579.61 sec*proc (1200 tests)
hard_fail                   = 628.82 sec*proc (11 tests)
intermittent_stat_fail      = 9619.82 sec*proc (131 tests)
nexus                       = 188.04 sec*proc (70 tests)
performance                 = 47808.84 sec*proc (128 tests)
poor_test                   =  30.18 sec*proc (14 tests)
quality_unknown             = 61722.83 sec*proc (1988 tests)
rmg                         =  16.22 sec*proc (5 tests)
unit                        = 217.32 sec*proc (46 tests)
unstable                    =  58.96 sec*proc (553 tests)

Total Test time (real) = 3910.09 sec

The following tests FAILED:
        402 - short-heg_14_gamma-sjb-4-4-totenergy (Failed)
        480 - short-LiH_ae-vmc_hf_noj_estimator_energydensity_voronoi-4-4-check (Failed)
        574 - short-LiH_pp-vmc_hf_sdj_estimator_spindensity-4-4 (Failed)
        575 - short-LiH_pp-vmc_hf_sdj_estimator_spindensity-4-4-check (Failed)
        577 - short-LiH_pp-vmc_hf_sdj_estimator_energydensity_voronoi-4-4-check (Failed)
        593 - short-H4-RHF-orb-opt-16-1-totenergy (Failed)
        597 - short-H4-orb-opt-1-16-totenergy (Failed)
        598 - short-H4-orb-opt-dmc-16-1 (Failed)
        599 - short-H4-orb-opt-dmc-16-1-totenergy (Failed)
        601 - short-H4-orb-opt-dmc-1-16-totenergy (Failed)
        617 - short-H4-batched-hybrid-16-1-totenergy (Failed)
        619 - short-H4-batched-hybrid-1-16-totenergy (Failed)
        636 - short-H4-msd-orb-opt-16-1-totenergy (Failed)
        754 - short-He_param_grad_legacy_driver-1-16-jud_0 (Failed)
        755 - short-He_param_grad_legacy_driver-1-16-jud_1 (Failed)
        760 - He_param_grad_load-1-16-jud_1 (Failed)
        911 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1 (Failed)
        912 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-totenergy_A (Failed)
        913 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-totenergy_B (Failed)
        914 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-dtotenergy_AB (Failed)
        915 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-ionion_A (Failed)
        916 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-ionion_B (Failed)
        917 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-dionion_AB (Failed)
        918 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-eeenergy_A (Failed)
        919 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-eeenergy_B (Failed)
        920 - short-bccH_1x1x1_ae-csvmc-pbyp-nodrift_sdj-16-1-deeenergy_AB (Failed)
        921 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1 (Failed)
        922 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-totenergy_A (Failed)
        923 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-totenergy_B (Failed)
        924 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-dtotenergy_AB (Failed)
        925 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-ionion_A (Failed)
        926 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-ionion_B (Failed)
        927 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-dionion_AB (Failed)
        928 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-eeenergy_A (Failed)
        929 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-eeenergy_B (Failed)
        930 - short-bccH_1x1x1_ae-csvmc-all-nodrift_sdj-16-1-deeenergy_AB (Failed)
        931 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1 (Failed)
        932 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-totenergy_A (Failed)
        933 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-totenergy_B (Failed)
        934 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-dtotenergy_AB (Failed)
        935 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-ionion_A (Failed)
        936 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-ionion_B (Failed)
        937 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-dionion_AB (Failed)
        938 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-eeenergy_A (Failed)
        939 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-eeenergy_B (Failed)
        940 - short-bccH_1x1x1_ae-csvmc-all_sdj-16-1-deeenergy_AB (Failed)
        1063 - short-diamondC_1x1x1_pp-vmcbatch-estimator-onebodydensitymatrices-4-4-check (Failed)
        1067 - short-diamondC_1x1x1_pp-vmc-noJ-estimator-1rdm-4-4-check (Failed)
        1069 - short-diamondC_1x1x1_pp-vmc-J2-estimator-1rdm-4-4-check (Failed)
        1071 - short-diamondC_1x1x1_pp-vmc-estimator-energydensity-cell-4-4-check (Failed)
        1073 - short-diamondC_1x1x1_pp-vmc-estimator-energydensity-voronoi-4-4-check (Failed)
        1075 - short-diamondC_1x1x1_pp-dmc-estimator-energydensity-cell-4-4-check (Failed)
Errors while running CTest
```
which was the result of
```
cmake -GNinja  -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx -DQMC_DATA=/scratch/pk7/QMC_DATA_FULL/ -DQMC_PERFORMANCE_NIO_MAX_ATOMS=64 ../qmcpack; ninja
ctest -j 32 -E long --output-on-failure 
```

This is well within bounds for a release and does not include core features, not that the short test failures are good:

- csvmc appears broken
- orb opt is the usual trouble, as the tests need revisiting.
- He_param_grad_load-1-16-jud_1  is bad(?)
- 4x4 observables are still bad (crash, missing implementation)
- HEG failed by 3.4 sigma; test might simply need revisiting.

**Comment by prckent** [comment]
1.44% coverage increase compared to 3.14

**Pull request merged**